kedro/config/omegaconf_config.py

```diff
@@ -109,6 +111,19 @@ def __init__(
         # It's easier to introduce them step by step, but removing them would be a breaking change.
         self._clear_omegaconf_resolvers()
 
+        file_mimetype, _ = mimetypes.guess_type(conf_source)
+        if not file_mimetype:
+            self._protocol = "file"
+        else:
+            if file_mimetype == "application/x-tar":
+                self._protocol = "tar"
+            # It's extremely complicated to create a zip file with a temporary directory, which
+            # we use in our unit tests. That's why the following two lines aren't covered in tests.
+            # The functionality has been tested manually.
+            elif file_mimetype == "application/zip":  # pragma: no cover
+                self._protocol = "zip"  # pragma: no cover
+        self._fs = fsspec.filesystem(protocol=self._protocol, fo=conf_source)
+
```

Comment:
https://github.com/fsspec/filesystem_spec/blob/master/fsspec/implementations/tests/test_tar.py

Is there something useful we can borrow from the `fsspec` test-suite here?